fdbclient/include/fdbclient/TenantManagement.actor.h

```diff
@@ -135,6 +136,17 @@ Future<std::pair<TenantMapEntry, bool>> createTenantTransaction(Transaction tr,
 	return std::make_pair(newTenant, true);
 }
 
+ACTOR template <class Transaction>
+Future<int64_t> getNextTenantId(Transaction tr) {
+	state typename transaction_future_type<Transaction, Optional<Value>>::type lastIdFuture = tr->get(tenantLastIdKey);
```

Comment:
I'm having trouble following this perfectly, but is there a reason the type declaration is now different from `lastIdFuture` before?

Reply:
The declaration below comes from the database object represented by the template parameter `DB`. From that, we can get the future type via the path `DB::TransactionT::FutureT`. The template parameter here (`Transaction`) does not afford us that ability because it is passed in as a `Reference` or a pointer. So instead, we use the `transaction_future_type` helper that extracts `FutureT` from the `Transaction` based on whether it is a plain object, a `Reference`, or a pointer.